src/lib/logstore/log_store_service.cpp

```diff
@@ -235,11 +237,6 @@ void LogStoreService::rollback_super_blk_found(const sisl::byte_view& buf, void*
         auto id = rollback_sb->logdev_id;
         LOGDEBUGMOD(logstore, "Log dev rollback superblk found logdev={}", id);
         const auto it = m_id_logdev_map.find(id);
-        if (it == m_id_logdev_map.end()) {
-            LOGERROR("logdev={} found but not opened yet, it will be discarded after logstore is started", id);
-            m_unopened_logdev.insert(id);
-        }
-
         if (it != m_id_logdev_map.end()) {
             logdev = it->second;
         } else {
```

Comment:
I am not sure whether it is valid to have a `rollback_superblk` but do not have a `logdev_superblock`  ?  
If not, maybe we dont need the L239-L245?
If yes,  what is the case?   do we need to add the logdev into `m_unopened_logdev` as well? 

cc @sanebay

Reply:
Yes dont understand the changes done in the logdev. If we find ether logdev or rollback superblk, we create logdev structure. If its nt opened, we delete them, thats a feature.

Reply:
@sanebay can you point out where we remove the logdev from `m_unopened_logdev` if it was opened?

Reply:
@sanebay 
if i understand correctly,  a rollback superblk aims to  record the rollback info of a logdev, right? a rollback superblk in only meaningful when it has an corresponding log dev instance.
so if we find a rollback superblk without the corresponding log dev, that should be an error.

in fact, i wanted to add and assert in line 246 to confirm each rollback superblk has a an corresponding log dev.

Reply:
> @sanebay can you point out where we remove the logdev from `m_unopened_logdev` if it was opened?
https://github.com/eBay/HomeStore/blob/78aead7920e60beaef29fa5f049177b3f03dadd9/src/lib/logstore/log_store_service.cpp#L100
PTAL!

Reply:
@JacksonYao287  I am asking, in previous code, where we remove OPENED logdev from `m_unopened_logdev`...

Reply:
I believe the position i mentioned above is the only place to remove element from m_unopened_logdev.

Reply:
with the ordering we enforced, do we still have the `else` clause ? 

maybe it is an assert?

Reply:
yes, that what I wanted, I believe we need assert here. pls take a look at my above comment.
>in fact, i wanted to add an assert in line 246 to confirm each rollback superblk has a an corresponding log dev.